src/edg.rs

```diff
@@ -372,28 +490,60 @@ impl<B: Broker<V> + Debug, G: ExtendedDependencyGraph<V> + Send + Sync + Debug,
             None => {
                 // UNEXPLORED
                 // Line 6
+                trace!(
+                    ?edge,
+                    ?weight,
+                    assignment = "UNEXPLORED",
+                    "processing negation edge"
+                );
                 self.add_depend(&edge.target, Edges::NEGATION(edge.clone()));
-                let (weight1, weight2) = weight
-                    .split()
-                    .unwrap_or_else(|_| panic!("Worker {} ran out of weight", self.id));
-                self.broker
-                    .send(self.id, Message::NEGATION(edge.clone(), weight1));
-                self.explore(&edge.target, weight2);
+                self.queue_negation(edge.clone(), weight.clone());
+                self.explore(&edge.target, weight.clone());
             }
             Some(assignment) => match assignment {
                 VertexAssignment::UNDECIDED => {
+                    trace!(?edge, ?weight, assignment = ?VertexAssignment::UNDECIDED, "processing negation edge");
                     self.final_assign(&edge.source, VertexAssignment::TRUE, weight)
                 }
                 VertexAssignment::FALSE => {
+                    trace!(?edge, ?weight, assignment = ?VertexAssignment::FALSE, "processing negation edge");
                     self.final_assign(&edge.source, VertexAssignment::TRUE, weight)
                 }
-                VertexAssignment::TRUE => self.delete_edge(Edges::NEGATION(edge), weight),
+                VertexAssignment::TRUE => {
+                    trace!(?edge, ?weight, assignment = ?VertexAssignment::TRUE, "processing negation edge");
+                    self.delete_edge(Edges::NEGATION(edge), weight)
+                }
             },
         }
     }
 
+    /// Queueing unsafe negation, which will be queued to negation channel whenever
+    /// release negation is called
+    fn queue_negation(&mut self, edge: NegationEdge<V>, weight: Weight) {
+        let mut len = self.unsafe_edges.len();
+        let mut dist = 0;
+        if let Some(n) = self.distances.get(&edge.source) {
+            dist = n.clone();
+        }
+        while len <= dist as usize {
+            self.unsafe_edges.push(Vec::new());
+            len = len + 1
```

Comment:
Fixes `clippy` warning
```suggestion
            len += 1;
```